timesketch/lib/analyzers/ioc_extraction.py

```diff
@@ -0,0 +1,280 @@
+            'placeholder': 'Partial Field Name',
+            'default_value': '',
+            'optional': True
+        },
+        {
+            'name': 'match_re',
+            'type': 'ts-dynamic-form-text-input',
+            'label': 'The regular expression to extract data from field',
+            'placeholder': 'Regular Expression',
+            'default_value': '([^a-zA-Z0-9]|^)$value$([^a-zA-Z0-9]|$)',
+            'optional': True
+        },
+        {
+            'name': 're_flags',
+            'type': 'ts-dynamic-form-multi-select-input',
+            'label': 'List of flags to pass to the regular expression',
+            'placeholder': 'Regular Expression flags',
+            'default_value': [],
+            'options': RE_FLAGS,
+            'optional': True
+        },
+        {
+            'name': 'store_as',
+            'type': 'ts-dynamic-form-text-input',
+            'label': 'Name of the field to store the extracted ioc in',
+            'placeholder': 'Store results as field name',
+            'default_value': 'ioc'
+        },
+        {
+            'name': 'tags',
+            'type': 'ts-dynamic-form-multi-select-input',
+            'label': 'Tag to add to events with matches',
+            'placeholder': 'Tag to add to events',
+            'default_value': '',
+            'optional': True
+        }
+    ]
+
+
+    def __init__(self, index_name, sketch_id, config=None):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+            config: Optional dict that contains the configuration for the
+                analyzer. If not provided, the default YAML file will be
+                loaded up.
+        """
+        self.index_name = index_name
+        super(IOCExtractionSketchPlugin, self).__init__(
+            index_name, sketch_id)
+        self._config = config
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result.
+        """
+        config = self._config or interface.get_yaml_config(self.CONFIG_FILE)
+        if not config:
+            return 'Unable to parse the config file.'
+        query, ret_fields, ioc_db = self.parse_config(config)
+        if not query or not ret_fields:
+            return 'Error to make query'
+        if not ioc_db:
+            return 'Error to parse config'
+        logger.info((
+            'Return field '
+            'error: {0!s}').format(ret_fields))
+        return self.extract_ioc(query, ret_fields, ioc_db)
+
+    def parse_config(self, ioc_config):
+        """Create IOC DB and make request
+
+        Args:
+            config: A dict that contains the configuration for the ioc
+                extraction. See data/ioc.yaml for fields and further
+                documentation of what needs to be defined.
+
+        Returns:
```

Comment:
so you can be more clear here... it returns a tuple with three entries, and then you go ahead and contine with the description you have here